### PR TITLE
Display a warning message when java version greater than 1.8 is used with Spring Boot 1.5.x and below

### PR DESCRIPTION
**Pull request by anjumfatima90**

### CODE REVIEW

File: dev/com.ibm.ws.springboot.support_fat/fat/src/com/ibm/ws/springboot/support/fat/Java11WithSpringBoot15Tests.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return SPRING_BOOT_15_APP_BASE;
+    }
+
+    @Test
+    public void expectWarningMsgAndAppFailureWithJava11AndSpringBoot159() throws Exception {
+        String currentJavaVersion = System.getProperty("java.version");
+        if (currentJavaVersion.startsWith("1.")) {
+            server.stopServer();
+            //Update the java version to 11
+            System.setProperty("java.version", "11");
```
Comment (tjwatson) [inline dev/com.ibm.ws.springboot.support_fat/fat/src/com/ibm/ws/springboot/support/fat/Java11WithSpringBoot15Tests.java]: How does this work, setting the java.version property here is for the FAT execution, that should not lead to the JVM used to launch the test server to think it is Java 11.  I think this test needs to be conditional to expect failure messages on Java 11 and success messages on Java 8.

Comment (anjumfatima90) [reply dev/com.ibm.ws.springboot.support_fat/fat/src/com/ibm/ws/springboot/support/fat/Java11WithSpringBoot15Tests.java]: Refactored the test class.

File: dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops [hunk 2]
```diff
@@ -80,3 +80,7 @@ warning.spring_config.ignored.session.useraction=Configure session timeout and s
 error.wrong.spring.boot.version=CWWKC0263E: The Spring Boot version {0} is not supported. A Spring Boot version within the range {1} is required.
 error.wrong.spring.boot.version.explanation=A Spring Boot version has been detected that is not supported by the server.
 error.wrong.spring.boot.version.useraction=Update the Spring Boot application to use a version of Spring Boot that is within the required version range.
+
+warning.java.version.not.supported=CWWKC0264W: The current Java version {0} is not supported by the Spring Boot version {1}. Upgrade the Spring Boot version of the application to 2.0.x or later.
+warning.java.version.not.supported.explanation=Java versions greater than 1.8 are not supported by the Spring Boot versions 1.5.x and below.
```
Comment (bsbyrd1) [inline dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops]: Suggestion:
Java versions higher than 1.8 ...

Comment (anjumfatima90) [reply dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops]: Changed

**Comment by anjumfatima90** [comment]
#build

**Comment by tjwatson** [comment]
The build shows three flavors of feature verifier errors that we need to pay attention to:

```
ERROR: com.ibm.websphere.appserver.springBootHandler-1.0
[BAD_PACKAGE_VERSION com.ibm.ws.app.manager.springboot.container@1.0.0]
Changes were detected in package com.ibm.ws.app.manager.springboot.container@1.0.0 which mean the package version is now incorrect. The baseline knows the package at version 1.0.0 Change the version of the package from its current version 1.0.0 to 2.0.0 Remember!! Major Changes in CD Open stream require POC Approval. If you do not have this, do not proceed with the changes that lead to this error.
For More Information, visit http://was.pok.ibm.com/xwiki/bin/view/Build/FeatureChecker
```

and

```
ERROR: com.ibm.websphere.appserver.springBootHandler-1.0
[MISSING_DEV_RESOURCE com.ibm.ws.app.manager.springboot.container.ApplicationError.class]
Path  com.ibm.ws.app.manager.springboot.container.ApplicationError.class is known to have changed for com.ibm.websphere.appserver.springBootHandler-1.0 but there is no associated dev resource. This error should not be ignored unless this is an issue with a legacy resource that is part of an API/SPI package but is not exposed as API/SPI.
For More Information, visit http://was.pok.ibm.com/xwiki/bin/view/Build/FeatureChecker
```

and

```
ERROR: com.ibm.websphere.appserver.springBootHandler-1.0
[PACKAGE_COMPARE_MAJOR com.ibm.ws.app.manager.springboot.container@null]

The class/interface ApplicationError$Type has been deleted, or has been changed to private or package default visibility
For More Information, visit http://was.pok.ibm.com/xwiki/bin/view/Build/FeatureChecker
```

Notice that we expose the following packages to applications for our spring boot autoconfiguration implementations to have access to:

```
IBM-API-Package: \
 com.ibm.ws.app.manager.springboot.container.config; type="internal", \
 com.ibm.ws.app.manager.springboot.container; type="internal"
```

It is curious that the `type="internal"` packages are checked like this.
@brenthdaniel Is that expected?  If we simply update the version to 2.0 for that package, will it be OK, or will the API checks still think we made a breaking change?

**Comment by brenthdaniel** [comment]
Updating the package version should be enough. The missing dev resource error may still have to be ignored, though

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
Java 11 build is at
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_XbteQNOyEeio4_O0Fzon5g

**Comment by anjumfatima90** [comment]
Build with CL feature checker ignores

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_hGumQNbMEei3nL3FFg5SFA

**Pull request merged**